components/dashboard/src/components/DropDown2.tsx

```diff
@@ -0,0 +1,153 @@
+        },
+        [props],
+    );
+    const [search, setSearch] = useState<string>("");
+    const filteredOptions = props.getElements(search);
+    const [selectedElementTemp, setSelectedElementTemp] = useState<string | undefined>(filteredOptions[0]?.id);
+
+    const onKeyDown = useMemo(
+        () => (e: React.KeyboardEvent) => {
+            if (!showDropDown) {
+                return;
+            }
+            if (e.key === "ArrowDown") {
+                e.preventDefault();
+                let idx = filteredOptions.findIndex((e) => e.id === selectedElementTemp);
+                while (idx++ < filteredOptions.length - 1) {
+                    const candidate = filteredOptions[idx];
+                    if (candidate.isSelectable) {
+                        setSelectedElementTemp(candidate.id);
+                        return;
+                    }
+                }
+                return;
+            }
+            if (e.key === "ArrowUp") {
+                e.preventDefault();
+                let idx = filteredOptions.findIndex((e) => e.id === selectedElementTemp);
+                while (idx-- > 0) {
+                    const candidate = filteredOptions[idx];
+                    if (candidate.isSelectable) {
+                        setSelectedElementTemp(candidate.id);
+                        return;
+                    }
+                }
+                return;
+            }
+            if (e.key === "Escape") {
+                setShowDropDown(false);
+                e.preventDefault();
+            }
+            if (e.key === "Enter" && selectedElementTemp && filteredOptions.some((e) => e.id === selectedElementTemp)) {
+                e.preventDefault();
+                props.onSelectionChange(selectedElementTemp);
+                setShowDropDown(false);
+            }
+        },
+        [filteredOptions, props, selectedElementTemp, showDropDown],
+    );
+
+    const onBlur = useMemo(
+        () => (e: React.FocusEvent) => {
+            setShowDropDown(false);
+        },
+        [setShowDropDown],
+    );
+
+    const doShowDropDown = useMemo(() => () => setShowDropDown(true), []);
+    return (
+        <div onKeyDown={onKeyDown} onBlur={onBlur} className="relative flex flex-col">
+            <div
+                className={
+                    "h-16 rounded-lg bg-gray-100 dark:bg-gray-800 flex items-center px-2 " +
+                    (!showDropDown && " hover:bg-gray-200 dark:hover:bg-gray-700 cursor-pointer")
+                }
+                onClick={doShowDropDown}
+            >
+                {props.children}
+                <div className="flex-grow" />
+                <div className="mr-2">
+                    <Arrow direction={showDropDown ? "up" : "down"} />
+                </div>
+            </div>
+            {showDropDown && (
+                <div className="absolute w-full top-10 bg-gray-100 dark:bg-gray-900 max-h-72 overflow-auto rounded-lg mt-3 z-50 p-2">
```

Comment:
**issue**: This approach can leak some rounded corners on the dark theme. Could we fix this with color changes or round sides separately when expanding or collapsing the dropdown? See [relevant docs](https://tailwindcss.com/docs/border-radius#rounding-sides-separately) in Tailwind.

![Frame 1223](https://user-images.githubusercontent.com/120486/208639937-e62770a0-92f0-459c-9999-aa24e6f23430.png)

Reply:
I've fixed this so that on expand the component and the drop down don't have rounded corners where they connect.